std/meta.zig

```diff
@@ -117,6 +117,44 @@ test "std.meta.Child" {
     testing.expect(Child(promise->u8) == u8);
 }
 
+/// Given an array/pointer type, return the slice type `[]Child`. Preserves `const`.
```

Comment:
These functions will be considered in another PR (https://github.com/ziglang/zig/pull/2782).  For now I just include a version of them to get the tests to work.